packages/lesswrong/components/linkPreview/FootnotePreview.tsx

```diff
@@ -132,10 +132,25 @@ const FootnotePreview = ({classes, href, id, rel, children}: {
   // links to anchors like "#fn:1" which will crash this because it has a ':' in
   // it.
   try {
-    // Grab contents of linked footnote if it exists
-    footnoteHTML = document.querySelector(href)?.innerHTML || "";
-    // Check whether the footnotehas nonempty contents
-    footnoteContentsNonempty = !!Array.from(document.querySelectorAll(`${href} p`)).reduce((acc, p) => acc + p.textContent, "").trim();
+    // `href` is (probably) an anchor link, of the form `#fn1234`. Since it starts
+    // with a hash it can also be used as a CSS selector, which finds its contents
+    // in the footer.
+    const footnoteContentsElement = document.querySelector(href);
+    footnoteHTML = footnoteContentsElement?.innerHTML || "";
+    
+    // Decide whether the footnote is nonempty. This is tricky because while there
+    // are consistently formatted footnotes created by our editor plugins, there
+    // are also wacky irregular footnotes present in imported HTML and similar
+    // things. Eg https://www.lesswrong.com/posts/ACGeaAk6KButv2xwQ/the-halo-effect
+    // We can't just condition on the footnote containing non-whitespace text,
+    // because footnotes sometimes have their number and backlink in a place that
+    // would be mistaken for their body. Our current heuristic is that a footnote
+    // is nonempty if it contains at least one <p> which contains non-whitespace
+    // text, which might false-negative on rare cases like an image-only footnote
+    // but which seems to work in practice.
+    footnoteContentsNonempty = !!footnoteContentsElement
+      && !!Array.from(footnoteContentsElement.querySelectorAll("p"))
+        .reduce((acc, p) => acc + p.textContent, "").trim();
```

Comment:
I feel like this reduce could be an `a.some`

Reply:
`.some(p => !!p.textContent.trim())`?

Reply:
(Agreed that these are equivalent and simpler; this line was preexisting and I only reformatted it without changing that part.)